### PR TITLE
Use db.Conn to fix postgres lock/unlock

### PR DESCRIPTION
**Pull request by chrisaxiom**
Hello @dhui , I would also like this one if possible merged in.

Thanks!

### CODE REVIEW

File: database/postgres/postgres.go [hunk 1]
```diff
@@ -33,7 +35,8 @@ type Config struct {
 }
 
 type Postgres struct {
-	db       *sql.DB
+	// Locking and unlocking need to use the same connection
+	db       *sql.Conn
```
Comment (dhui) [inline database/postgres/postgres.go]: Rename `db` to `conn`

File: database/postgres/postgres.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// +build go1.9
```
Comment (dhui) [inline database/postgres/postgres.go]: We need to [support Go 1.8](https://golang.org/doc/devel/release.html#policy) until Go 1.10 is released in [Feb 2018](https://tip.golang.org/doc/go1.10#introduction)

So we can either:
1. Wait until Go 1.10 is release and we can drop Go 1.8 support and remove this build tag
1. Or have a separate implementation of postgres.go which supports Go 1.8. See [net/context](https://github.com/golang/net/tree/master/context) for an example of a package supporting multiple Go versions

**Comment by chrisaxiom** [comment]
I will perhaps remake this PR myself with another file for 1.9.  This is not my branch so I am actually unsure how to resolve the conflicts anyways.   I will incorporate your request to rename `db` to `conn` as well.  Thanks for the quick response!

**Comment by dhui** [comment]
Fixed in release [v3.2.0](https://github.com/golang-migrate/migrate/releases/tag/v3.2.0)

**Pull request closed without merging**